Assets/Scripts/GraphService.cs

```diff
@@ -62,53 +62,56 @@ public void SetGraph(Graph graph)
     public bool IsNode(GameObject gameObject) => FindNodeByGameObject(gameObject) != null;
 
     public Node FindNodeByGameObject(GameObject gameObject) =>
-      Graph.nodes.SingleOrDefault(n => n.gameObject == gameObject);
+      Graph.Nodes.SingleOrDefault(n => n.gameObject == gameObject);
 
     public bool IsEdge(GameObject gameObject) => FindEdgeByGameObject(gameObject) != null;
 
     public Edge FindEdgeByGameObject(GameObject gameObject) =>
-      Graph.edges.SingleOrDefault(n => n.gameObject == gameObject);
+      Graph.Edges.SingleOrDefault(n => n.segmentGroup.Contains(gameObject));
 
-    public Node FindNodeById(int id) => Graph.nodes.SingleOrDefault(n => n.id == id);
+    public Node FindNodeById(int id) => Graph.Nodes.SingleOrDefault(n => n.Id == id);
 
     public void AddNode(Vector3 position, Quaternion rotation)
```

Comment:
I think we need an issue to completely overhaul node/edge creation as they've become sa complicated and spread out that I don't see how this code could be maintained going forward

Reply:
True, I am trying to think of a better design while refactoring but it's not that easy